src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx

```diff
@@ -162,9 +164,41 @@ export const PupilViewsProgrammeListing = ({
           baseSlug={baseSlug}
           chosenFactors={chosenFactors}
           programmes={programmes}
-          onClick={(f) =>
-            setChosenFactors({ ...chosenFactors, [currentFactor]: f })
-          }
+          onClick={(f) => {
+            setChosenFactors({
+              ...chosenFactors,
+              [currentFactor]: f,
+            });
+          }}
+          onCallback={(f) => {
+            const activeFilters: {
```

Comment:
As per my other comment. This can live in the onClick callback. Also this pretty lengthy so it would be nice to refactor into a helper function with descriptive title